ignite/handlers/checkpoint.py

```diff
@@ -353,6 +370,36 @@ def _setup_checkpoint(self) -> dict:
             checkpoint[k] = obj.state_dict()
         return checkpoint
 
+    @staticmethod
+    def setup_filename_pattern(
+        filename_prefix: str = "",
+        with_score: bool = False,
+        with_score_name: bool = False,
+        with_global_step_transform: bool = False,
+    ) -> str:
```

Comment:
@Joel-hanson a docstring for `setup_filename_pattern` is missing. Could you please add one. Thanks!

Reply:
Thinking more about the argument `filename_prefix` of `setup_filename_pattern`. Seems like it is a bit misleading as its value is not used. As we only check if it is defined or not I propose to replace it by `with_prefix`.

Reply:
@vfdev-5 will this work?
```python
"""Helper method to get the default filename pattern for a checkpoint.
Args:
    with_prefix (bool): If True, the `filename_prefix` with an underscore (filename_prefix_) will be added to the filename pattern.
        By default the `filename_prefix` will not be appended to the filename pattern
    with_score (bool): If True, this indicates that the score function is not provided, then it will look for ``with_score_name``
        and ``with_global_step_transform`` to make the required filename pattern. If it is False then this indicates that the score
        function is provided.
    with_score_name (bool): If True, this indicates that the score name is not provided, then it will look for 
        ``with_global_step_transform`` to make the required filename pattern. If it is False then this indicates that the score
        name is provided.
    with_global_step_transform (bool): If True, this indicates that there is no global step, then it will make the
        required filename pattern accordingly. If it is False then this indicates that there is a global step.
Example:
    .. code-block:: python

        from ignite.handlers import Checkpoint

        filename_pattern = Checkpoint.setup_filename_pattern("model", True, True, True)
        print(filename_pattern)
        > "{filename_prefix}_{name}_{score}.{ext}"
"""
```

Reply:
Yes, looks good. Nit: Make all default values to True and in the example `Checkpoint.setup_filename_pattern("model", True, True, True)` -> `Checkpoint.setup_filename_pattern()`

Reply:
@vfdev-5 I have updated the requested changes.